pilot/pkg/proxy/envoy/v2/config.go

```diff
@@ -357,14 +354,15 @@ func mayApplyInboundAuth(listener *xdsapi.Listener, mesh *meshconfig.MeshConfig,
 
 // buildTCPListener constructs a listener for the TCP proxy
 // in addition, it enables mongo proxy filter based on the protocol
+// NOTE: The TCP listeners setup so far will not work as we are not setting up tcp routes properly
```

Comment:
We do use TCP proxy in the pilot proxy - can be used as a model to generate same config (in particular the TLS part)

Reply:
Yes, but we don't use the full route match in tcp proxy. We put in some sensible default in Envoy, when I was doing v2 filter transition.. It will send to the upstream cluster if its just a simple port match.. when you have multiple services on same TCP port, you need the filter chain match (which is not done) or the old deprecated_v1